### DISCUSSION TITLE
Adventure map. Animation of map objects when viewing quick info dialog or other dialog windows.

### DISCUSSION THREAD

**Discussion by Districh-ru** [ihhub/fheroes2#6856] [Ideas]
Currently in fheroes2, like in original game, when the game is showing a message window, in example quick info, the animation of all map objects stops:
![fheroes2 2023-03-17 21-21-46-313](https://user-images.githubusercontent.com/113276641/225992644-e6850fa0-674c-4772-81e2-7ab656055d35.gif)

Should we resume this animation for some cases of dialog windows?
Or it will distract the user from reading the shown dialog..?

![fheroes2 2023-03-17 21-25-52-817](https://user-images.githubusercontent.com/113276641/225992896-85f5bcf8-abd2-4b1e-918f-e0311480c622.gif)

**Comment by Mr-Bajs** [comment]
Letting the animations continue simply looks neater. 
I think hardly the objects animation distracts either way. Also in the shown picture both the portal and the altars animation still continues, having some objects stop and some to animate is rather inconsistent.

**Comment by LeHerosInconnu** [comment]
Hello @Districh-ru,

Personally I wouldn't be disturbed if the adventure map was animated when an information window is displayed, on the contrary.
I vote for continuous animation of the adventure map if it is not too complicated to implement.

**Comment by Districh-ru** [comment]
Hello @LeHerosInconnu,
For the shown Quick Info dialog it is quite simple, but for some other it could be a little complicated.
Quick info dialog for castle and hero can be called from the Castle or View Kingdom. For these cases the should be no adventure map render including animation. 
Right-clicking on the hero portrait or castle icon shows the hero/castle position on mini-map and this will force the Adventure map to change the focus. But these difficulties can be overcome. (Also maybe showing clicked hero/castle on Adventure map will help player to easily plan his turn and do a quick view of his heroes positions).

**Reply by LeHerosInconnu** [reply]
Hello @Districh-ru,

> Right-clicking on the hero portrait or castle icon shows the hero/castle position on mini-map and this will force the Adventure map to change the focus. But these difficulties can be overcome. (Also maybe showing clicked hero/castle on Adventure map will help player to easily plan his turn and do a quick view of his heroes positions).

Would the adventure map view before the maintained right click be repositioned to its initial position when the right click is released?
It is important that this is the case.

Assuming that the program will have new features later on.
Which is better from a programming point of view (less complicated, takes less time, etc.), making a change and adapting that change as the new features come in, or waiting to have the new features first and making that change later?

**Reply by Districh-ru** [reply]
Hello @LeHerosInconnu,
Yes, it will be repositioned to the initial position - like the cursor on mini-map.
The easiest way is to change the focus tor the time when right mouse button is pressed as the map position is already changed for repositioning the mini-map cursor (the Adventure map does not update so it preserves its position).
I plan to implement this improvement after a one other PR.

Hello @ihhub, @Branikolog, @oleg-derevenetz, @zenseii. What do you think, should we implement such improvement?

**Reply by oleg-derevenetz** [reply]
I suggest implementing this ONLY if there is a way to do it without over-complicating the rendering logic. Overcomplication is always a bad thing because it's hard to maintain overly complex logic and hard to extend it later (especially for project newcomers).

**Reply by Districh-ru** [reply]
Hi @oleg-derevenetz. For the simple Quick View (not for castles and heroes) currently all the changes are: cache the Quick View dialog with the text in a sprite (render to a sprite instead of to display), call Interface::GameArea::Redraw() like it is done for Adventure map render and render cached sprite over it.
If the implementation for other dialogs will assume a complicated rendering approach I'll don't do it.

**Reply by Branikolog** [reply]
Hi, @Districh-ru .

>I suggest implementing this ONLY if there is a way to do it without over-complicating the rendering logic. Overcomplication is always a bad thing because it's hard to maintain overly complex logic and hard to extend it later (especially for project newcomers).

I agree with @oleg-derevenetz .
